test/test_bwd.py

```diff
@@ -29,15 +29,18 @@ def __init__(self):
         self.mm_param0 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
         self.mm_param1 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
         self.mm_param2 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
+        self.register_buffer("cval", torch.randn((d_hid,), requires_grad=False))
         self.lin1 = torch.nn.Linear(d_hid, d_hid)
         self.lin2 = torch.nn.Linear(d_hid, d_hid)
 
     def forward(self, x):
         x = torch.mm(x, self.mm_param0)
         x = torch.relu(x)
+        a_constant = self.cval.clone()
         pipe_split()
         x = torch.mm(x, self.mm_param1)
-        skip = x
+        # try passing a value that doesn't require_grad across skip boundaries
+        skip = x + a_constant
```

Comment:
Should we do `skip = a_constant` instead? 
(Or pass `a_constant` across the boundary)

Reply:
Just for my dumb brain, if `skip = x + a_constant`, would `skip` requires_grad? (I think it will, so we won't be testing this issue, is that correct?)

Reply:
i posted an updated version of the test code. i validated that it catches both the tuple issue and the grad issue.

`skip + a_constant` would require_grad, which is why i kept them separate initially, but i found a way